src/otx/algo/visual_prompting/segment_anything.py

```diff
@@ -116,29 +125,100 @@ def load_checkpoint(
                 f"To manually load {load_from}, try to set it to trainer.checkpoint.",
             )
 
-    def forward(
+    def forward(self, mode: str, *args, **kwargs) -> Any:  # noqa: ANN401
+        """Forward method for visual prompting task."""
+        assert mode in ["finetuning", "learn", "infer"]  # noqa: S101
```

Comment:
why do we need to have separated finetuning and learn for segment_anything?

Reply:
As far as I understood, learn = finetuning, infer = pseudo-labeling wo prompt in segment_anything (w/ point decoder), while learn=ref_feat extraction, infer=pseudo-labeling w/ prompt in zero_shot_segment_anything. Please correct me if I was wrong.

Reply:
`finetuning` uses original SAM forward with batch-wise prompts. `learn` also uses original SAM forward, but it uses prompt-wise prompts. `infer` uses PerSAM forward pipeline. So, splitting forward pipeline is required.